data/keystore/src/main/java/tech/pegasys/artemis/bls/keystore/KeyStoreLoader.java

```diff
@@ -0,0 +1,104 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.artemis.bls.keystore.model.KdfParam;
+import tech.pegasys.artemis.bls.keystore.model.KeyStoreData;
+
+/** Provide utility methods to load/store BLS KeyStore from json format */
+public class KeyStoreLoader {
+  private static final Logger LOG = LogManager.getLogger();
+  private static final ObjectMapper OBJECT_MAPPER =
+      new ObjectMapper().registerModule(new KeyStoreBytesModule());
+
+  public static KeyStoreData loadFromFile(final Path keystoreFile)
+      throws KeyStoreValidationException {
+    checkNotNull(keystoreFile, "KeyStore path cannot be null");
+
+    try {
+      final KeyStoreData keyStoreData =
+          OBJECT_MAPPER.readValue(keystoreFile.toFile(), KeyStoreData.class);
+      keyStoreData.validate();
+      return keyStoreData;
+    } catch (final KeyStoreValidationException e) {
+      throw new KeyStoreValidationException("Error in parsing keystore: " + e.getMessage(), e);
+    } catch (final JsonParseException e) {
+      throw new KeyStoreValidationException("Error in parsing keystore: Invalid Json format", e);
+    } catch (final JsonMappingException e) {
+      throw convertToKeyStoreValidationException(e);
+    } catch (final IOException e) {
+      LOG.error("Error in parsing keystore: " + e.getMessage());
+      throw new KeyStoreValidationException("Error in parsing keystore: " + e.getMessage(), e);
+    }
+  }
+
+  private static KeyStoreValidationException convertToKeyStoreValidationException(
+      final JsonMappingException e) {
+    final String cause;
+    if (e.getCause() instanceof KeyStoreValidationException) {
+      cause = e.getCause().getMessage();
+    } else if (e instanceof InvalidTypeIdException) {
+      cause = getKdfFunctionErrorMessage((InvalidTypeIdException) e);
+    } else {
+      cause = "Missing required json elements";
+    }
+    return new KeyStoreValidationException(
+        String.format("Error in parsing keystore: %s", cause), e);
+  }
+
+  private static String getKdfFunctionErrorMessage(final InvalidTypeIdException e) {
+    if (e.getBaseType().getRawClass() == KdfParam.class) {
+      return "Kdf function [" + e.getTypeId() + "] is not supported.";
+    }
+    return "Missing required json elements";
+  }
+
+  public static void saveToFile(final Path keystoreFile, final KeyStoreData keyStoreData) {
+    checkNotNull(keystoreFile, "KeyStore path cannot be null");
+    checkNotNull(keyStoreData, "KeyStore data cannot be null");
+
+    try {
+      Files.writeString(keystoreFile, toJson(keyStoreData), StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      throw new KeyStoreValidationException(
+          "Error in writing KeyStore to file: " + e.getMessage(), e);
+    }
+  }
+
+  public static String toJson(final KeyStoreData keyStoreData) {
+    checkNotNull(keyStoreData, "KeyStore data cannot be null");
```

Comment:
Looks like this could be private and then it wouldn't need the `checkNotNull` (no need to recheck things we pass ourselves).

Reply:
yes, this is now private.